### DISCUSSION TITLE
Namer plugin chaining

### DISCUSSION THREAD

**Discussion by eirikb** [parcel-bundler/parcel#6075] [Q&A] [answered]
Hi

I've noticed `namers` in **.parcelrc** accepts an array, how an I use this?
Or rather, how can my `Namer`-plugin pass the naming over to the next namer when I don't want to change the name?

In my case I want to rename *.mustache.*-files to not have "mustache" in them.  
[My code](https://github.com/eirikb/parcel-namer-mustache/blob/main/src/index.ts) fails sometimes when `bundleGraph.getBundleGroupsContainingBundle(bundle)` is `undefined`. 

I had a look at https://github.com/ol-loginov/parcel-namer-rewrite, and there they allow chaining of **namer-default** by manually calling out to it.

**Comment by mischnic** [comment, marked as answer]
> pass the naming over to the next namer when I don't want to change the name?

https://v2.parceljs.org/plugin-system/namer/:

> If some namer plugin returns null, the next namer in the list is queried (and so on).

**Reply by eirikb** [reply]
Thank you!

Once again I made a silly mistake. I tried returning `null`, but at one place I had optional chaining in TS and it ended up being `undefined`. The error message I got said as much, but emphasized on return being string (null not mentioned), so I was confused.  
Sorry for the mess and great answer as usual :smile: 

OT: Wonder why the library I linked to simply doesn't do the same

**Reply by mischnic** [reply]
> Wonder why the library I linked to simply doesn't do the same

Maybe because it first executes the default namer and then applies these rewrites on the result?

**Reply by eirikb** [reply]
Yes, that could be
